frontend/src/components/employees/index.tsx

```diff
@@ -117,7 +119,9 @@ const renderTableRow = (tableColumns: TableColumn[]) => (row: StaffData) => (
 const renderTable = (tableColumns: TableColumn[]) => (data: StaffData[]) => (
   <Table size="small">
     {renderTableHeaders(tableColumns)}
-    <TableBody>{data.map(renderTableRow(tableColumns))}</TableBody>
+    <TableBody>
+      {data.length > 0 && data.map(renderTableRow(tableColumns))}
```

Comment:
Won't this give
`<TableBody>false</TableBody>` if data.length is 0?
`map` already handles empty lists just fine.

Reply:
I didn't change this, must have been Iain?

Although I believe this notation will perform a check to see if there is any data. If there is no data, then it will not perform the map and show nothing. I think I've used this before to stop an error caused by data being undefined/empty.

Reply:
OK, I've learnt something. This is a logical statement which returns `false` when data.length is 0, but if false is passed as an element's child in JSX it is displayed as an empty tag
https://react-cn.github.io/react/tips/false-in-jsx.html

I still don't see why this guard would be necessary, which edge case does it guard against?
`map` handles `[]` just fine and typescript should prevent any use of `undefined` or `null` (which will break anyway when you try to access the object's length).

We can leave it for now and ask Iain when he's back.

Reply:
Yeah I was thinking that - I've done a check before but just done, for example:

```{data && data.value}```

to ensure that data.value isn't displayed until `data` is present to prevent undefined errors. But yeah ok I'll leave it and we can discuss later.